### ISSUE TITLE
bug(cdkTextareaAutosize): Animation of height calculates default height wrongly

### ISSUE THREAD

**Issue by henrikdahl8240** [angular/components#18999]
I have a component, A, which contains a textarea, which has the attribute cdkTextareaAutosize.

When I animate the height of A from 0 to * (i.e. the default height). The height of A is animated to the wrong final value which appears like the default height assumes that the height of the textarea is only one line and not the number of lines it actually has.

As soon as the animation completes, the height jumps to the right height.

Therefore I guess, that there is a kind of synchronization problem between computing the height of the textarea by the cdkTextareaAutosize attribute and determining the default height for '*'.

Any hint?

#### Environment

- Angular: 8.2.14.
- CDK/Material: 8.2.3.
- Browser(s): Google Chrome.
- Operating System (e.g. Windows, macOS, Ubuntu):  Windows 10 Pro

**Comment by crisbeto** [comment]
Can you post a Stackblitz that shows the issue? I wasn't able to reproduce it based on the provided info.

**Comment by henrikdahl8240** [comment]
Yes. Try to see here: https://stackblitz.com/edit/angular-afhkfy

As you can see, there are two create buttons and two close buttons, one at each header. The Save buttons only shows an announcement.

It's obviously the easiest to see, if you click at "Open in New Window".

It is very easy to see the problem I addressed, isn't it?

So what do you say?

**Comment by crisbeto** [comment]
I can see the issue, but I'm not sure what it is off the top of my head. It looks like the animations module measures the height of the element before the textarea has had the time to resize. You might be able to work around it by adding a delay to your animation.

**Comment by henrikdahl8240** [comment]
That's what I meant with "synchronization problem".

I have tried to add 300ms break: "animate(1000, ..." => "animate('1000ms 300ms', ...".

As you can see, it now gives a completely wrong result.

The Angular CDK sample has code like:
```
    triggerResize() {
        // Wait for changes to be applied, then trigger textarea resize.
        this._ngZone.onStable.pipe(take(1))
            .subscribe(() => {
                this.autosize.resizeToFitContent(true);
            });
    }

```
Should this method be invoked e.g. by the event handler when starting the animation or the height has already been calculated at this moment?

May you recommend some other place to invoke this method for solving the problem? It doesn't bring any improvement when invoking it in OnInit() or AfterViewInit().

Do you think, that this "error" may be corrected in the Angular CDK?

**Comment by crisbeto** [comment]
Waiting for the zone to stabilize could still be too late for your case. Have you tried without the `zone.onStable` subscription?

**Comment by henrikdahl8240** [comment]
Yes, the sample does not include the triggerResize() method and no other invocation of resizeToFitContent(...), so the sample already fulfills what you ask for.

Do you think, that the problem may be fixed in the Angular CDK, because it's probably where it should be done. User's will obviously assume that something has not been programmed correctly with this strange jumping around, so it givex exactly the opposite effect than the intention with the animation.

**Comment by crisbeto** [comment]
We are closing this issue, because we've determined, using a semi-automated process, that it is no longer relevant, has low engagement, or it does not have enough information to proceed. Feel free to comment if you think the issue should be reopened, or open a new issue with the latest version of our code instead.
